src/core/src/op/fake_convert.cpp

```diff
@@ -4,17 +4,47 @@
 
 #include "openvino/op/fake_convert.hpp"
 
+#include "element_visitor.hpp"
 #include "itt.hpp"
+#include "openvino/reference/convert.hpp"
+#include "openvino/reference/fake_convert.hpp"
 
 namespace ov {
 namespace op {
 namespace v13 {
-namespace fake_convert {
+namespace fake_convert_details {
 static const std::vector<std::string>& get_valid_types() {
     static const std::vector<std::string> valid_types{"f8e4m3", "f8e5m2"};
     return valid_types;
 }
-}  // namespace fake_convert
+
+struct Evaluate : element::NoAction<bool> {
+    using element::NoAction<bool>::visit;
+    template <element::Type_t ET, class T = fundamental_type_for<ET>>
+    static result_type visit(ov::TensorVector& outputs,
+                             const ov::TensorVector& inputs,
+                             const std::string& destination_type) {
+        if (inputs.size() == 2) {  // Default shift
```

Comment:
It would be better do this check before call template and if some input is optional make dummy tensor reburied by reference impl.

Reply:
Considered as a follow up code refactor.